tests/fips/openssl/openssl_fips_alglist.pm

```diff
@@ -15,64 +15,75 @@ use base "consoletest";
-    if (!is_sle('<15-sp4') && ($current_ver ge 1.1.1)) {
-        record_info('openssl version', "Version of Current openssl package: $current_ver");
-    }
-    else {
-        record_soft_failure('jsc#SLE-19640: openssl version is outdated and need to be updated over 1.1.1+ for SLE15-SP4');
-    }
-
-    # Seperate the diffrent openssl command usage between SLE12 and SLE 15-SP{0..5} and SLE 15-SP6+
-    if (is_sle('<15')) {
-        # List message digest algorithms in fips mode
-        # only SHA1 and SHA2 (224, 256, 384, 512) are approved in fips mode
-        # Note: DSA is short of DSA-SHA1, so it is also valid item
+sub test_openssl1 {
+    my $openssl_binary = shift // "openssl";
+    eval {
         validate_script_output
-"echo -n 'Invalid Hash: '; openssl list-message-digest-algorithms | sed -e '/SHA1/d' -e '/SHA224/d' -e '/SHA256/d' -e '/SHA384/d' -e '/SHA512/d' -e '/DSA/d' | wc -l",
+"echo -n 'Invalid Hash: '; $openssl_binary list -digest-algorithms | sed -e '/SHA1/d' -e '/SHA224/d' -e '/SHA256/d' -e '/SHA384/d' -e '/SHA512/d' -e '/DSA/d' -e '/SHA3-224/d' -e '/SHA3-256/d' -e '/SHA3-384/d' -e '/SHA3-512/d' -e '/SHAKE128/d' -e '/SHAKE256/d' | wc -l",
           sub { m/^Invalid Hash: 0$/ };
 
-        # List public key algorithms in fips mode
-        # only RSA, DSA, ECDSA, EC DH, CMAC and HMAC are approved in fips mode
         validate_script_output
-"echo -n 'Invalid Pubkey: '; openssl list-public-key-algorithms | grep '^Name' | sed -e '/RSA/d' -e '/rsa/d' -e '/DSA/d' -e '/dsa/d' -e '/EC/d' -e '/DH/d' -e '/HMAC/d' -e '/CMAC/d' | wc -l",
+"echo -n 'Invalid Pubkey: '; $openssl_binary list -public-key-algorithms | grep '^Name' | sed -e '/RSA/d' -e '/rsa/d' -e '/DSA/d' -e '/dsa/d' -e '/EC/d' -e '/DH/d' -e '/HMAC/d' -e '/CMAC/d' | wc -l",
           sub { m/^Invalid Pubkey: 0$/ };
-    } elsif (is_sle('<15-SP6')) {
-        eval {
-            validate_script_output
-"echo -n 'Invalid Hash: '; openssl list -digest-algorithms | sed -e '/SHA1/d' -e '/SHA224/d' -e '/SHA256/d' -e '/SHA384/d' -e '/SHA512/d' -e '/DSA/d' -e '/SHA3-224/d' -e '/SHA3-256/d' -e '/SHA3-384/d' -e '/SHA3-512/d' -e '/SHAKE128/d' -e '/SHAKE256/d' | wc -l",
-              sub { m/^Invalid Hash: 0$/ };
+    }
+}
 
-            validate_script_output
-"echo -n 'Invalid Pubkey: '; openssl list -public-key-algorithms | grep '^Name' | sed -e '/RSA/d' -e '/rsa/d' -e '/DSA/d' -e '/dsa/d' -e '/EC/d' -e '/DH/d' -e '/HMAC/d' -e '/CMAC/d' | wc -l",
-              sub { m/^Invalid Pubkey: 0$/ };
-        }
-    } else {
-        eval {
-            validate_script_output
+sub test_openssl3 {
+    eval {
+        validate_script_output
 "echo -n 'Invalid Hash: '; openssl list -digest-algorithms | grep fips | sed -e '/SHA1/d' -e '/SHA224/d' -e '/SHA256/d' -e '/SHA384/d' -e '/SHA512/d' -e '/DSA/d' -e '/SHA3-224/d' -e '/SHA3-256/d' -e '/SHA3-384/d' -e '/SHA3-512/d' -e '/SHAKE128/d' -e '/SHAKE256/d' -e '/KECCAK/d' | wc -l",
```

Comment:
looks like a rather long command. Maybe would be better to store the invalid hashes in a Perl array and check the unfiltered output if any matches ?

Reply:
good point! I'll update the code